### DISCUSSION TITLE
Custom Components & Containers

### DISCUSSION THREAD

**Discussion by andrew-bauer-calibrated** [microsoft/aspire#1188] [Q&A] [answered]
How do I go about creating either custom aspire components or importing custom containers? Where do I find the documentation on it?

For example, I would like to use ArangoDB in my project. I would like to configure my aspire project to use the ArangoDB container image and inject the endpoint into projects.

Our organisation uses ArangoDB on multiple client projects, so I would like to create a set of helper component & packages for it (similar to the supplied AddRedisContainer component & Aspire.StackExchange.Redis.OutputCaching package).

**Comment by davidfowl** [comment, marked as answer]
There's no document as yet (cc @IEvangelist) but you should be able to read the code for one of the components to create your own. This is a good example of adding a new component and resource for mongodb https://github.com/dotnet/aspire/pull/986
